repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -1479,20 +1479,28 @@ private void validateEntityAssociations(String guid, List<AtlasClassification> c
         List<String>    entityClassifications = getClassificationNames(guid);
         String          entityTypeName        = AtlasGraphUtilsV2.getTypeNameFromGuid(graph, guid);
         AtlasEntityType entityType            = typeRegistry.getEntityTypeByName(entityTypeName);
+        Set<String> processedTagTypeNames = new HashSet<>();
 
-        for (AtlasClassification classification : classifications) {
-            String newClassification = classification.getTypeName();
+        List <AtlasClassification> copyList = new ArrayList<>(classifications);
+        for (AtlasClassification classification : copyList) {
 
-            if (CollectionUtils.isNotEmpty(entityClassifications) && entityClassifications.contains(newClassification)) {
-                throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "entity: " + guid +
-                        ", already associated with classification: " + newClassification);
-            }
+            if (processedTagTypeNames.contains(classification.getTypeName())){
+                classifications.remove(classification);
```

Comment:
Why is this required?

Reply:
There is flaw in this method.
consider request payload with multiple elements but same combination of "entityGuid" & "displayName"/"typeName"

It this case, as this method checks for associations in graph & passes the validation
Then the same classification gets attached to a same entity multiple times

This change handles this flaw by maintaining processed classification names in the loop